
Some fixup to get the sparc assembly built on sparc64.

  - Rene Rebe <rene@exactcode.de>

diff -ur --exclude '*.o' --exclude Makefile xc/extras/Mesa/src/mesa/sparc/clip.S xc-sparc64/extras/Mesa/src/mesa/sparc/clip.S
--- xc/extras/Mesa/src/mesa/sparc/clip.S	2004-06-16 09:19:23.000000000 +0000
+++ xc-sparc64/extras/Mesa/src/mesa/sparc/clip.S	2005-01-23 05:44:01.000000000 +0000
@@ -25,12 +25,10 @@
 #define VEC_SIZE_3   	7
 #define VEC_SIZE_4   	15
 
-#ifdef SVR4
-        /* Solaris requires this for 64-bit. */
+        /* Solaris/Linux requires this for 64-bit. */
         .register %g2, #scratch
         .register %g3, #scratch
         .register %g7, #scratch
-#endif
 
 	.text
 	.align		64
diff -ur --exclude '*.o' --exclude Makefile xc/extras/Mesa/src/mesa/sparc/glapi_sparc.S xc-sparc64/extras/Mesa/src/mesa/sparc/glapi_sparc.S
--- xc/extras/Mesa/src/mesa/sparc/glapi_sparc.S	2004-06-16 09:19:24.000000000 +0000
+++ xc-sparc64/extras/Mesa/src/mesa/sparc/glapi_sparc.S	2005-01-23 06:14:10.000000000 +0000
@@ -6,6 +6,9 @@
  */
 
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 .align 32
 .globl __glapi_sparc_icache_flush
diff -ur --exclude '*.o' --exclude Makefile xc/extras/Mesa/src/mesa/sparc/norm.S xc-sparc64/extras/Mesa/src/mesa/sparc/norm.S
--- xc/extras/Mesa/src/mesa/sparc/norm.S	2004-06-16 09:19:24.000000000 +0000
+++ xc-sparc64/extras/Mesa/src/mesa/sparc/norm.S	2005-01-23 05:44:22.000000000 +0000
@@ -2,11 +2,9 @@
 
 #include "sparc_matrix.h"
 
-#ifdef SVR4
-	/* Solaris requires this for 64-bit. */
+	/* Solaris/Linux requires this for 64-bit. */
 	.register %g2, #scratch
 	.register %g3, #scratch
-#endif
 
 	.text
 
diff -ur --exclude '*.o' --exclude Makefile xc/extras/Mesa/src/mesa/sparc/xform.S xc-sparc64/extras/Mesa/src/mesa/sparc/xform.S
--- xc/extras/Mesa/src/mesa/sparc/xform.S	2004-06-16 09:19:25.000000000 +0000
+++ xc-sparc64/extras/Mesa/src/mesa/sparc/xform.S	2005-01-23 05:43:38.000000000 +0000
@@ -18,11 +18,9 @@
 
 #include "sparc_matrix.h"
 
-#ifdef SVR4
-	/* Solaris requires this for 64-bit. */
+	/* Solaris/Linux requires this for 64-bit. */
         .register %g2, #scratch
         .register %g3, #scratch
-#endif
 
 	.text
 	.align	64
diff -ur --exclude '*.o' --exclude Makefile xc/programs/Xserver/hw/xfree86/drivers/sunffb/VISmoveImage.s xc-sparc64/programs/Xserver/hw/xfree86/drivers/sunffb/VISmoveImage.s
--- xc/programs/Xserver/hw/xfree86/drivers/sunffb/VISmoveImage.s	2004-04-23 19:48:16.000000000 +0000
+++ xc-sparc64/programs/Xserver/hw/xfree86/drivers/sunffb/VISmoveImage.s	2005-01-23 05:37:34.000000000 +0000
@@ -131,6 +131,9 @@
 #define load_bank2		f48-f62
 #define fnop			fmovd	%ftmp2, %ftmp2
 
+        .register %g2,#scratch
+        .register %g3,#scratch
+
 #define LOAD(f32,f46,tgt,tgtr)							\
 	subcc			%curw, 64, %curw;				\
 	bleu,pn			%ICC, tgt;					\
diff -ur --exclude '*.o' --exclude Makefile xc/programs/Xserver/hw/xfree86/drivers/sunffb/ffb_asm.s xc-sparc64/programs/Xserver/hw/xfree86/drivers/sunffb/ffb_asm.s
--- xc/programs/Xserver/hw/xfree86/drivers/sunffb/ffb_asm.s	2004-04-23 19:48:16.000000000 +0000
+++ xc-sparc64/programs/Xserver/hw/xfree86/drivers/sunffb/ffb_asm.s	2005-01-23 06:00:57.000000000 +0000
@@ -54,6 +54,9 @@
 #define SEG_X2		0x04
 #define SEG_Y2		0x06
 
+        .register %g2,#scratch
+        .register %g3,#scratch
+
 	.text
 
 	.align	32
diff -ur --exclude '*.o' --exclude Makefile xc/programs/Xserver/hw/xfree86/loader/loader.c xc-sparc64/programs/Xserver/hw/xfree86/loader/loader.c
--- xc/programs/Xserver/hw/xfree86/loader/loader.c	2004-08-29 21:06:00.000000000 +0000
+++ xc-sparc64/programs/Xserver/hw/xfree86/loader/loader.c	2005-01-23 16:15:46.000000000 +0000
@@ -114,7 +114,7 @@
 static int refCount[MAX_HANDLE];
 #endif
 
-#if defined(__sparc__) && defined(__GNUC__) && !defined(__FreeBSD__)
+#if defined(__sparc__) && defined(__GNUC__) && !defined(__FreeBSD__) && !defined(linux)
 # define SYMFUNCDOT(func) { "." #func, (funcptr)&__sparc_dot_ ## func },
 # if !defined(__OpenBSD__)
 # define SYMFUNCDOT89(func) { "." #func, (funcptr)&func ## _sparcv89 },
@@ -343,7 +343,7 @@
     LoaderAddSymbols(-1, -1, dixLookupTab);
     LoaderAddSymbols(-1, -1, fontLookupTab);
     LoaderAddSymbols(-1, -1, extLookupTab);
-#if defined(__sparc__) && !defined(__FreeBSD__)
+#if defined(__sparc__) && !defined(__FreeBSD__) && !defined(linux)
 #ifdef linux
     if (sparcUseHWMulDiv())
 	LoaderAddSymbols(-1, -1, SparcV89LookupTab);
